.github/workflows/tests.yaml

```diff
@@ -0,0 +1,86 @@
+name: Test
+
+permissions:
+  contents: read
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    name: "Tests (${{ matrix.compiler }}, C++${{ matrix.standard}}, ${{ matrix.mode }}, ${{ matrix.enables }})"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # only the compilers supported by setup-cpp
+        compiler: [clang++-18, gcc-13]
+        standard: [20, 23]
+        mode: [dev, debug, release]
+        include:
+          # only build this combination with dpdk enabled, so we don't double
+          # the size of the test matrix, and can at least test the build with
+          # dpdk enabled.
```

Comment:
A thought:
If we're already building 12 times, would it hurt us to build 14 times - with one *extra* build for DPDK and one *extra* build for modules - instead of modifying one of the existing build combinations?
Or am I misunderstanding and what this code says is to add a 13th and 14th build combination? If that's the case, the comment can be improved to say that.

Reply:
the latter. we already have [clang++-18, gcc-13] x [20, 23] x [dev, debug, release] = 12 combinations. i am adding another two. sure, i am revising the comment.